inc/Editor/BlockManagement/ConfigRegistry.php

```diff
@@ -48,84 +90,116 @@ public static function register_block( array $block_config = [] ): bool|WP_Error
+				return self::create_error( $block_title, sprintf( 'Display query "%s" not found', $display_query_key ) );
+			}
+		}
 
-		// Build the base configuration for the block. This is our own internal
-		// configuration, not what will be passed to WordPress's register_block_type.
-		// @see BlockRegistration::register_block_type::register_blocks.
-		$config = [
-			'description' => '',
-			'icon' => $block_config['icon'] ?? 'cloud',
-			'instructions' => $block_config['instructions'] ?? null,
-			'name' => $block_name,
-			'overrides' => $block_config['overrides'] ?? [],
-			'patterns' => [],
-			'queries' => [
-				self::DISPLAY_QUERY_KEY => $display_query,
-			],
-			'selectors' => [
-				[
-					'image_url' => $display_query->get_image_url(),
+		$queries = [];
+		$selectors = [];
+		$display_queries_to_selectors_map = [];
+
+		// Generate the selectors for the selector queries.
+		foreach ( $block_config[ self::QUERIES_KEY ] as $query_key => $query ) {
+			// Inflate the query, and add it to the queries array.
+			$query = self::inflate_query( $query );
+			$queries[ $query_key ] = $query;
+
+			$input_schema = $query->get_input_schema();
+			$output_schema = $query->get_output_schema();
+
+			// Generate the selector for the display query, and then continue on to the next query.
+			if ( in_array( $query_key, $block_config['display_queries'], true ) ) {
+				$is_collection = true === ( $output_schema['is_collection'] ?? false );
+				$has_required_variables = array_reduce(
+					array_column( $input_schema, 'required' ),
+					fn( $carry, $required ) => $carry || ( $required ?? true ),
+					false
+				);
+
+				$selectors[] = [
+					'image_url' => $query->get_image_url(),
 					'inputs' => self::map_input_variables( $input_schema ),
 					'name' => $has_required_variables ? 'Manual input' : ( $is_collection ? 'Load collection' : 'Load item' ),
-					'query_key' => self::DISPLAY_QUERY_KEY,
+					'query_key' => $query_key,
 					'type' => $has_required_variables ? 'manual-input' : 'load-without-input',
-				],
-			],
-			'title' => $block_title,
-		];
+				];
 
-		// Register "selectors" which allow the user to use a query to assist in
-		// selecting data for display by the block.
-		foreach ( $block_config[ self::SELECTION_QUERIES_KEY ] ?? [] as $selection_query ) {
-			$from_query = self::inflate_query( $selection_query['query'] );
-			$from_query_type = $selection_query['type'];
-			$to_query = $display_query;
+				$display_queries_to_selectors_map[ $query_key ][] = $query_key;
 
-			$config['queries'][ $from_query::class ] = $from_query;
+				continue;
+			}
 
-			$from_input_schema = $from_query->get_input_schema();
-			$from_output_schema = $from_query->get_output_schema();
+			// Infer the type of the query, for selector generation and to validate the non-display queries.
+			$inferred_type = self::infer_query_type( $input_schema, $output_schema );
+			if ( is_wp_error( $inferred_type ) ) {
```

Comment:
This inverses the previous way of throwing an error for invalid query based on the type. Now, based on the inputs the type is inferred. So if the inputs aren't setup correctly, then the type won't be inferred and the error would be thrown.